src/pages/participants.js

```diff
@@ -4,188 +4,394 @@ import { FaTwitter } from "react-icons/fa";
 import Helmet from "../components/helmet";
 import Navbar from "../components/navbar";
 
-export default () => (
-  <div>
-    <Helmet />
-    <Navbar />
-    <section className="section">
-      <h1 className="title has-text-centered">Participant registration</h1>
-    </section>
-    <section className="hero is-primary is-bold is-medium">
-      <div className="hero-body">
-        <div className="container">
-          <h1 className="title has-text-centered">
-            We are sorry but the registration is not open yet!
-          </h1>
-          <div className="buttons is-centered">
-            <div className="button is-medium is-link is-in-hero">
-              <span className="icon">
-                <FaTwitter size="fa" />
-              </span>
-              <a
-                href="https://twitter.com/pylondiniumgir1/"
-                target="_blank"
-                rel="noopener noreferrer"
-              >
-                Keep connected
-              </a>
-            </div>
-          </div>
-        </div>
-      </div>
-    </section>
-    <section className="hero is-primary is-bold is-medium is-hidden-touch is-hidden-tablet">
-      <div className="hero-body">
-        <div className="container">
-          <h1 className="title has-text-centered">The registration is open!</h1>
-        </div>
-      </div>
-    </section>
-    <section className="section is-hidden-touch is-hidden-tablet">
-      <div className="columns is-centered">
-        <div className="column is-half">
+
+function Request(form) {
+  function validate() {
+    // Validate first name
+    let first_name = formData.get("first_name");
+    inputErrors.first_name = first_name.toLowerCase() === first_name.toUpperCase();
+
+    // Validate last name
+    let last_name = formData.get("last_name");
+    inputErrors.last_name = last_name.toLowerCase() === last_name.toUpperCase();
+
+    // Validate email
+    let email_regex = /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/;
+    inputErrors.email = !email_regex.test(formData.get("email"));
+
+    // It is compulsory to select an os
+    inputErrors.os = formData.getAll("os").length === 0;
+
+    // It is compulsory to select a level of experience
+    let experience = formData.getAll("experience");
+    inputErrors.experience = experience.length === 0;
+
+    // If the experience is not beginner, it is compulsory to say something
+    // about this experience
+    if (experience.filter(elem => elem !== "beginner").length !== 0) {
+      inputErrors.experience_notes = formData.get("experience_notes") === "";
+    } else {
+      inputErrors.experience_notes = false;
+    }
+
+    // It is required to explain the motivations for the workshop
+    inputErrors.motivations = formData.get("motivations") === "";
+
+    // It is required to read and accept the coc
+    inputErrors.coc = formData.get("coc") !== "accept";
+
+    return !Object.keys(inputErrors).some((key) => inputErrors[key]);
```

Comment:
Using a functional style here obscures the purpose of the code. A more traditional, but much less exciting, loop will provide the same result, while being immediately obvious.